joomla/joomla-cms
# [4.0][WIP] Field password as Custom Element

Part of the effort to convert all fields to Custom Elements

### Summary of Changes
- One script to rule them all
- Ability to set multiple validation handlers (per field)
- Remove Bootstrap/Font Awesome and other unneeded dependencies so the field can play nicely in every known css framework
-  Some code improvements


@ciar4n can you take care of the css @todos ?

### Testing Instructions

For easy testing you can apply this patch and check the admin login:
<img width="488" alt="screen shot 2017-12-30 at 17 47 02" src="https://user-images.githubusercontent.com/3889375/34455906-a34ee950-ed89-11e7-92a9-b56697e8b9d0.png">
<img width="482" alt="screen shot 2017-12-30 at 17 47 12" src="https://user-images.githubusercontent.com/3889375/34455908-a7dffd7e-ed89-11e7-8ad1-d4be3611e8ad.png">
<img width="369" alt="screen shot 2017-12-30 at 17 47 20" src="https://user-images.githubusercontent.com/3889375/34455909-ab53768e-ed89-11e7-96bc-73bfe9a0ac7f.png">
<img width="386" alt="screen shot 2017-12-30 at 17 47 28" src="https://user-images.githubusercontent.com/3889375/34455910-afe180ce-ed89-11e7-8dc0-b03d01f47d0b.png">

**Issues**
- The show/hide button doesn't look like a button
- Still Bootstrap specific
- Still Font awesome specific
- Needs A11Y audit 
- Needs final approval form the design team

But the functionality should be spot on!
### Expected result



### Actual result



### Documentation Changes Required
